src/rust/multi.xkcd/src/languages/mod.rs

```diff
@@ -0,0 +1,110 @@
+use aidoku::{
+	error::Result,
+	prelude::format,
+	helpers::uri::encode_uri_component,
+	std::{
+		net::{HttpMethod, Request},
+		String, Vec,
+	},
+	Page,
+};
+use alloc::{borrow::ToOwned, vec};
+
+pub mod en;
+pub mod es;
+pub mod fr;
+pub mod ko;
+pub mod ru;
+pub mod zh;
+
+static THUMBNAIL_URL: &str = "https://fakeimg.pl/550x780/ffffff/6e7b91/?font=museo&text=xkcd";
+
+#[derive(PartialEq, Eq)]
+enum ImageVariant {
+	Latin,
+	Cjk,
+}
+
+trait ToImageUrl {
+	fn to_image_url(&self, variant: ImageVariant) -> String;
+}
+
+impl<S> ToImageUrl for S
+where
+	S: AsRef<str>,
+{
+	fn to_image_url(&self, variant: ImageVariant) -> String {
+		let text = encode_uri_component(self.as_ref());
+		match variant {
+            ImageVariant::Latin => format!("https://fakeimg.pl/1500x2126/ffffff/000000/?font=noto&font_size=42&text={text}"),
+            ImageVariant::Cjk => format!("https://placehold.jp/42/ffffff/000000/1500x2126.png?css=%7B%22padding%22%3A%22300px%22%2C%22text-align%22%3A%22left%22%7D&text={text}"),
```

Comment:
If you put padding on all sides of the text it ends up shifted downwards, so I think just doing it on the left and right is better.

also why is indentation using spaces here is this a rust format thing
```suggestion
            ImageVariant::Cjk => format!("https://placehold.jp/42/ffffff/000000/1500x2126.png?css=%7B%22padding%22%3A%22%200%20300px%22%2C%22text-align%22%3A%22left%22%7D&text={text}"),
```